shared/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package shared
```

Comment:
Re: the `shared` package and naming... 

'Shared'/'util'-type things tend to become dumping grounds. How about making this a root-level file called `deps.go` instead? `shared.interfaces` is a bit ambiguous. We're likely to have internal interfaces defined with Ringpop that probably shouldn't go in this file. I think what the interfaces in this file represents best are interfaces for Ringpop's 3rd party dependencies.